token/program/src/state.rs

```diff
@@ -331,9 +331,18 @@ pub trait GenericTokenAccount {
     }
 }
 
+/// The offset of state field in Account's C representation
+pub const ACCOUNT_INITIALIZED_INDEX: usize = 108;
+
+/// Check if the account data buffer represents an initialized account.
+/// This is checking the `state` (AccountState) field of an Account object.
+pub fn is_initialized_account(account_data: &[u8]) -> bool {
+    account_data.len() >= Account::LEN && account_data[ACCOUNT_INITIALIZED_INDEX] != 0
```

Comment:
Since we're doing the length check earlier in `valid_account_data`, perhaps this should just do the check at `ACCOUNT_INITIALIZED_INDEX`, same way as in token-2022, ie:

```
1 == *account_data.get(ACCOUNT_INITIALIZED_INDEX).unwrap_or(&0)
```

This will also reduce some of the repeated `len()` checks.